pkg/pillar/types/dns.go

```diff
@@ -287,50 +295,41 @@ func (status *DeviceNetworkStatus) MostlyEqualStatus(status2 DeviceNetworkStatus
 	return true
 }
 
-// GetPortByIfName - Get Port Status for port with given Ifname
-func (status *DeviceNetworkStatus) GetPortByIfName(
+// LookupPortByIfName returns status for port with the given interface name.
+func (status *DeviceNetworkStatus) LookupPortByIfName(
 	ifname string) *NetworkPortStatus {
 	for i := range status.Ports {
-		if status.Ports[i].IfName == ifname {
-			return &status.Ports[i]
+		port := &status.Ports[i]
+		if port.IfName == ifname {
+			return port
 		}
 	}
 	return nil
 }
 
-// GetPortsByLogicallabel - Get Port Status for all ports matching the given label.
-func (status *DeviceNetworkStatus) GetPortsByLogicallabel(
-	label string) (ports []*NetworkPortStatus) {
-	// Check for shared labels first.
-	switch label {
-	case UplinkLabel:
-		for i := range status.Ports {
-			if status.Version >= DPCIsMgmt && !status.Ports[i].IsMgmt {
-				continue
-			}
-			ports = append(ports, &status.Ports[i])
-		}
-		return ports
-	case FreeUplinkLabel:
-		for i := range status.Ports {
-			if status.Version >= DPCIsMgmt && !status.Ports[i].IsMgmt {
-				continue
-			}
-			if status.Ports[i].Cost > 0 {
-				continue
-			}
-			ports = append(ports, &status.Ports[i])
+// LookupPortByLogicallabel returns port configuration referenced by the logical label.
+func (status *DeviceNetworkStatus) LookupPortByLogicallabel(
+	label string) *NetworkPortStatus {
+	for i := range status.Ports {
+		port := &status.Ports[i]
+		if port.Logicallabel == label {
+			return port
 		}
-		return ports
 	}
-	// Label is referencing single port.
+	return nil
+}
+
+// LookupPortsByLabel returns status for every port which has the given label assigned
```

Comment:
Will you introduce usage of this function in a later PR?

Reply:
Yes, the next PR will be the main one for this feature and everything should make sense.